### PR TITLE
IXDTF time zone support

### PR DESCRIPTION
**Pull request by samchen61661**
IXDTF time zone support

### CODE REVIEW

File: experimental/ixdtf/src/parser.rs [hunk 1]
```diff
@@ -69,6 +73,7 @@ pub struct ParsedDateTime {
     pub minute: Option<u8>,
     pub second: Option<u8>,
     pub nano_second: Option<i32>,
+    pub time_zone: Option<Vec<u8>>,
```
Comment (samchen61661) [inline experimental/ixdtf/src/parser.rs:76]: Want to support `Option<[u8]>` but don't know how to do it.

Comment (Manishearth) [reply experimental/ixdtf/src/parser.rs:76]: you need to add a lifetime to the struct, and then the type becomes `Option<&'a [u8]>`

Comment (samchen61661) [reply experimental/ixdtf/src/parser.rs]: sg!

File: experimental/ixdtf/src/parser.rs [hunk 2]
```diff
@@ -91,6 +96,208 @@ impl<'a> DateTimeParser<'a> {
         return DateTimeParser { bytes };
     }
```
[the remark this answers is not in the document]
Comment (samchen61661) [reply experimental/ixdtf/src/parser.rs:98]: It turned out that I don't need to keep track of `substr_start` and `substr_len` in the `DateTimeParser` struct.

**Comment by robertbastian** [comment]
https://github.com/unicode-org/icu4x/pull/5349

**Pull request closed without merging**